x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportPutTrainedModelAction.java

```diff
@@ -249,10 +251,10 @@ protected void masterOperation(
                     configToReturn.getModelId(),
                     modelPackageConfigHolder.get(),
                     request.isWaitForCompletion(),
-                    ActionListener.wrap(
-                        downloadTriggered -> listener.onResponse(new PutTrainedModelAction.Response(configToReturn)),
-                        listener::onFailure
-                    )
+                    ActionListener.wrap((downloadTriggered) -> {
+                        listener.onResponse(new PutTrainedModelAction.Response(configToReturn));
+                        verifyMlNodesAndModelArchitectures(configToReturn, client, threadPool);
```

Comment:
Is it a problem to call listener.onResponse before preceding? I can imagine that the lines after listener.onResponse might never run if the listener ends the function. However, in this case it doesn't seem to have any impact.

Reply:
Yes, that is a problem. The code needs to be restructured so that `listener.onResponse` is called within the listener inside `verifyMlNodesAndModelArchitectures`. As things stand the response will never contain the warning.

It's probably best to make `verifyMlNodesAndModelArchitectures` take a listener as an extra argument. Then that will be called when the check completes.